core/chaincode/shim/chaincode.go

```diff
@@ -572,11 +572,11 @@ func (stub *ChaincodeStub) GetRows(tableName string, key []Column) (<-chan Row,
 	if err != nil {
 		return nil, fmt.Errorf("Error fetching rows: %s", err)
 	}
-	defer iter.Close()
 
 	rows := make(chan Row)
 
 	go func() {
+		defer iter.Close()
 		for iter.HasNext() {
 			_, rowBytes, err := iter.Next()
 			if err != nil {
```

Comment:
That doesn't sound like this PR is the root cause, but rather the fact that the iterators are asynchronous and/or concurrently created, to begin with.  If your program was working prior to this patch, it sounds like it was relying on the race with the (clearly incorrect) premature close of the first iterator.  My question is: why does the second iterator seemingly rely on the first being closed before it will function properly?  This sounds like the real bug.